ludwig/automl/auto_tune_config.py

```diff
@@ -0,0 +1,164 @@
+import copy
+from collections import OrderedDict
+
+import psutil
+import ray
+
+try:
+    import GPUtil
+except ImportError:
+    raise ImportError(
+        ' GPUtil is not installed. '
+        'In order to use this module please run '
+        'pip install GPUtil'
+    )
+
+from ludwig.api import LudwigModel
+from ludwig.automl.utils import get_available_resources
+from ludwig.data.preprocessing import preprocess_for_training
+from ludwig.features.feature_registries import update_config_with_metadata
+from ludwig.utils.defaults import merge_with_defaults
+from ludwig.constants import COMBINER, HYPEROPT, BATCH_SIZE, TRAINING, TYPE, PREPROCESSING, SPACE
+
+# maps variable search space that can be modified to minimum permissible value for the range
+RANKED_MODIFIABLE_PARAM_LIST = {
+    'tabnet': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.size': 8,
+        'combiner.output_size': 8,
+    }),
+    'concat': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.fc_size': 64,
+        'combiner.num_fc_layers': 1,
+
+    }),
+    'tabtransformer': OrderedDict({
+        'training.batch_size': 32,
+        'combiner.num_heads:': 4,
+        'combiner.output_size': 8,
+        'combiner.num_layers': 4,
+        'combiner.num_fc_layers': 1,
+    }),
+}
+
+
+BYTES_TO_MB = 1e6
```

Comment:
`nvidia-smi` (which is what GPUtil is using under the hood) reports in "MiB" (which is actually the correct way to measure MB, but I digress). So the correct conversion is:

```
BYTES_TO_MB = 1024 * 1024
```